### PR TITLE
Add TextInput as accessibility role

### PR DESCRIPTION
**Pull request by Montel**

### CODE REVIEW

File: internal/common/enums.rs [hunk 1]
```diff
@@ -345,6 +345,8 @@ macro_rules! for_each_enums {
                 Text,
                 /// The element is a [`ProgressIndicator`](../widgets/progressindicator.md) or behaves like one.
                 ProgressIndicator,
+                /// The role for a [`TextInput`](elements.md#TextInput) element.
```
Comment (ogoffart) [inline internal/common/enums.rs:348]: Should that be set on the TextInput or rather on the LineEdit/TextEdit?
It seems it is the later.
So I think this documentation should be fixed:

```suggestion
                /// The role for widget with editable text such as a 
                /// [`LineEdit`](../widgets/lineedit.md) or a [`TextEdit`](../widgets/textedit.md)
```

**Review by ogoffart — COMMENTED** [review]
Thanks for the patch. This looks good.

The role should also be added on the TextEdit widget.

**Review by ogoffart — APPROVED** [review]
Looks good to me.

@tronical does it looks ok. Note that this role set to a widget that has a TextInput inside, but not to TextInput itself.

This PR also doesn't set the properties accessible-label (or accessible-value?) and accessible-has-focus which would probably be required to work with the accessible interface.

**Comment by Montel** [comment]
accessible-has-focus  is automatic no ?
As we can't see it in source code

git grep "accessible-has-focu"
docs/reference/src/language/builtins/elements.md:70:-   **`accessible-has-focus`** (_in_ _bool_): Set to true when the current element currently has the focus.

**Comment by Montel** [comment]
Now I can see textedit/lineedit in accerciser application now.

**Comment by tronical** [comment]
>  Note that this role set to a widget that has a TextInput inside, but not to TextInput itself.

In the accessibility tree then there would be the LineEdit, and that would have another child (TextInput). In theory this could be matched to the TextField <> InlineTextBox hierarchy for example access kit needs (see #2895), but since the TextField is the one that needs to have the focus, I don't think that it'll work.

But this probably doesn't hurt - anything towards #2895 helps, and we can change it without breaking anything I think.

**Comment by ogoffart** [comment]
Regarding `accessible-has-focus` is is indeed in the documentation, but i see no reference to it in the code. @hunger do you remember anything about this?

**Pull request merged**